### PR TITLE
[Python] Apply feedback in base files

### PR DESCRIPTION
**Pull request by GiulianoDolceTalianoSantoro**
### Description

This PR fix PEP8 conventions, change non-descriptive variable names to increase readability and change constant variable names from lowercase to upper case on the files that have been modified during the DateTime library migration.

### Changes made

The changes were made on the following files:

- base_dateperiod.py
- base_datetime.py
- base_datetimeperiod.py
- base_duration.py
- base_holiday.py
- base_merged.py
- base_set.py
- base_time.py
- base_timeperiod.py
- date_time_extractor.py
- utilities.py
- date_time/utilities.py

### Testing

Below you will find a screenshot of the current tests status:

66497849-7e34ea80-ea93-11e9-8e62-3ac82d5fc4aa

### CODE REVIEW

**Review by Konigstiger — APPROVED** [review]
LGTM

**Review by ceciliaavila — APPROVED** [review]
LGTM 👍

**Review by LCarbonellRod — APPROVED** [review]
LGTM

File: Python/libraries/recognizers-number-with-unit/recognizers_number_with_unit/number_with_unit/models.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def __init__(self, extractor_parser: List[ExtractorParserModel]):
         self.extractor_parser: List[ExtractorParserModel] = extractor_parser
 
     def parse(self, query: str) -> List[ModelResult]:
-        query = QueryProcessor.preprocess(query, True)
+        query = QueryProcessor.pre_process(query, True)
```
Comment (Aliandi) [inline Python/libraries/recognizers-number-with-unit/recognizers_number_with_unit/number_with_unit/models.py]: this naming change is not ok

**Review by Aliandi — CHANGES_REQUESTED** [review]

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_set.py [hunk 2]
```diff
@@ -172,12 +172,13 @@ def match_each(self, extractor: DateTimeExtractor, source: str, reference: datet
 
         for match in regex.finditer(self.config.set_week_day_regex, source):
             trimmed_source = source[0:match.start(
-            )] + RegExpUtility.get_group(match, 'weekday') + source[match.end():]
+            )] + RegExpUtility.get_group(match, Constants.WEEKDAY_GROUP_NAME) + source[match.end():]
```
Comment (Maihav) [inline Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_set.py]: Can we fix this line? Should be better to open and close the source brackets at the same line; like this source[0:match.start()]

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_set.py [hunk 3]
```diff
@@ -435,7 +436,7 @@ def parse_each(self, extractor: DateTimeExtractor, parser: DateTimeParser,
         match = regex.search(self.config.set_week_day_regex, source)
         if match:
             trimmed_text = source[0:match.start(
-            )] + RegExpUtility.get_group(match, 'weekday') + source[match.end():]
+            )] + RegExpUtility.get_group(match, Constants.WEEKDAY_GROUP_NAME) + source[match.end():]
```
Comment (Maihav) [inline Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_set.py]: Same as above

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time.py [hunk 4]
```diff
@@ -301,36 +302,36 @@ def match_to_time(self, match: Match, reference: datetime):
             min_str = RegExpUtility.get_group(match, 'min')
```
Comment (Maihav) [inline Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time.py]: Here we have to replace the 'min' string by its constant: min_str = RegExpUtility.get_group(match, Constants.MIN_NUM)

Comment (Maihav) [reply Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time.py]: Same on line 339. It should be: time_prefix = RegExpUtility.get_group(match, Constants.PREFIX_GROUP_NAME)

**Pull request merged**